lib/languageFeatures/documentHightlightHandler.ts

```diff
@@ -1,25 +1,67 @@
 import { DocumentHighlight, DocumentHighlightKind, TextDocumentPositionParams } from 'vscode-languageserver';
-import { initialization, linters } from '../language-server';
-import { IHasLexerToken, implementsIHasLexerToken, OWrite } from '../parser/objects';
+import { IHasLexerToken, IHasReferenceToken, implementsIHasLexerToken, implementsIHasReferenceToken, ObjectBase, OWrite } from '../parser/objects';
+import { VhdlLinter } from '../vhdl-linter';
+import { findObjectFromPosition } from './findObjectFromPosition';
 
-export async function documentHighlightHandler(params: TextDocumentPositionParams): Promise<DocumentHighlight[] | null> {
-    await initialization;
-    const linter = linters.get(params.textDocument.uri);
-    if (!linter) {
-        return null;
-    }
+export function documentHighlightHandler(linter: VhdlLinter, params: TextDocumentPositionParams): DocumentHighlight[]  {
     const startI = linter.getIFromPosition(params.position);
-    const candidates = linter.file?.objectList.filter(object => object.range.start.i <= startI && startI <= object.range.end.i) ?? [];
-    candidates.sort((a, b) => (a.range.end.i - a.range.start.i) - (b.range.end.i - b.range.start.i));
-    const candidate = candidates.filter(candidate => implementsIHasLexerToken(candidate))[0] as IHasLexerToken;
-    if (typeof candidate === 'undefined') {
-        return null;
+
+    const candidates = findObjectFromPosition(linter, params.position)
+        .filter(candidate => {
+            // Only use Objects that have a lexer Token (either directly or as a reference Token)
+            // Also check if the cursor is actually in that token itself, or just in that objects range.
+            if (implementsIHasLexerToken(candidate)) {
+                if (candidate.lexerToken.range.start.i <= startI && startI <= candidate.lexerToken.range.end.i) {
+                    return true;
+                }
+            } else if (implementsIHasReferenceToken(candidate)) {
+                if (candidate.referenceToken.range.start.i <= startI && startI <= candidate.referenceToken.range.end.i) {
+                    return true;
+                }
+            }
+            return false;
+
+        }) as (ObjectBase & (IHasLexerToken | IHasReferenceToken))[]
+
+    const candidate = candidates[0];
+    if (candidate === undefined) {
+        return [];
+    }
```

Comment:
`candidate` can only be undefined when there are no candidates? It's easier to read when returning `[]` if `candidates.length === 0`